Sources/Commands/SwiftTool.swift

```diff
@@ -203,6 +204,17 @@ public class SwiftTool<Options: ToolOptions> {
         if let workspace = _workspace {
             return workspace
         }
+        let envSearchPaths = Utility.getEnvSearchPaths(
+            pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
+
+        guard let gitLocation = Utility.lookupExecutablePath(filename: Git.tool, searchPaths: envSearchPaths) else {
+            throw SwiftToolError.missingGitDependency
+            // NOTE(heckj): differentiate error reporting between not finding it and it not being executable?
+        }
+        guard localFileSystem.isExecutableFile(gitLocation) else {
+            throw SwiftToolError.missingGitDependency
+        }
```

Comment:
Thats a good point but I still think this belongs in Process class, we can cache the results so we don't need to check in subsequent calls. I'll leave it upto you if you want to implement he cache right now (it can just be a simple dictionary).